### PR TITLE
Simple branch

### PR DESCRIPTION
**Pull request by ndraper2**
Also known as simple graph.

### CODE REVIEW

File: simple_graph.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+class Graph(object):
+
+    def __init__(self):
+        self.graph = {}
+
+    def nodes(self):
+        l = []
+        for k, v in self.graph.iteritems():
+            l.append(k)
+        return l
```
Comment (cewing) [inline simple_graph.py:10]: why iterate over the items if all you want is the keys?  Is there no way to get just the keys by themselves without doing all this work?

File: simple_graph.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    def edges(self):
+        l = []
+        for k, v in self.graph.iteritems():
+            for edge in v:
+                l.append([k, edge])
```
Comment (cewing) [inline simple_graph.py:16]: only complaint is about naming, is `edge` really an "edge" here?  It's just another node isn't it?

File: simple_graph.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    def add_edge(self, node1, node2):
+        try:
+            self.graph.setdefault(node2, [])
+            self.graph[node1].append(node2)
+        except KeyError:
+            self.graph.setdefault(node1, [node2])
+
+    def del_node(self, node):
+        self.graph.pop(node, None)
```
Comment (cewing) [inline simple_graph.py:30]: What about edges that lead from other nodes to this one?

File: simple_graph.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    def del_edge(self, node1, node2):
+        temp = ''
+        try:
+            temp = self.graph[node1]
+        except KeyError:
+            raise KeyError("{} does not exist".format(node1))
+        try:
+            temp.remove(node2)
+        except KeyError:
```
Comment (cewing) [inline simple_graph.py:40]: Is this the right Exception type?

File: test_simple_graph.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+
+def test_add_edge(populated):
+    populated.add_edge(7, 5)
+    assert [7, 5] in populated.edges()
+
+
+def test_del_node(populated):
+    populated.del_node(5)
+    assert 5 not in populated.graph
```
Comment (cewing) [inline test_simple_graph.py:46]: You need more complex tests here.  you haven't got near enough tests to cover possible edge cases for deleting a node, which is why the problems I note above aren't breaking your tests.  Write tests that expose the problems before you fix this.

File: test_simple_graph.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+
+def test_del_edge(populated):
+    populated.del_edge(5, 7)
+    assert [5, 7] not in populated.edges()
```
Comment (cewing) [inline test_simple_graph.py:51]: Again, not enough tests.  the problem with the exception type above is not exposed here.  Write a test that exposes it before you fix it.
